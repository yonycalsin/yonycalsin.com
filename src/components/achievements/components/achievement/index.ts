export { default as Achievement } from './achievement'
