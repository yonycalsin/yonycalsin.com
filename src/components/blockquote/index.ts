export { default as Blockquote } from './blockquote'
