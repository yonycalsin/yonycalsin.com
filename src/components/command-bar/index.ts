export * from './command-bar'
export * from './command-bar-start-action'
