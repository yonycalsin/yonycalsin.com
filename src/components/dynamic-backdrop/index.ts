export { default as DynamicBackdrop } from './dynamic-backdrop'
