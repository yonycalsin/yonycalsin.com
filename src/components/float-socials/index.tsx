import * as React from 'react'
import { Link, VStack } from '@chakra-ui/react'

import { analyticsEvents } from '~/analytics/events'
import { analytics } from '~/analytics/google-analytics'
import { socialLinks } from '~/utils/constants'

/**
 * @deprecated this component will be remove in the future
 */
export const FloatSocials = () => {
  const onEventClick = (social: string) => {
    analytics.event({
      action: analyticsEvents.CLICK_IN_FLOAT_SOCIAL_LINK,
      category: 'redirect-external-link',
      label: social,
    })
  }

  return (
    <VStack
      display={{
        base: 'none',
        lg: 'flex',
      }}
      spacing="5"
      position="fixed"
      bottom="8"
      left="8"
    >
      <Link target="_blank" rel="noreferrer" href={socialLinks.GITHUB} onClick={() => onEventClick('github')}>
        Github
      </Link>
      <Link target="_blank" rel="noreferrer" href={socialLinks.LINKEDIN} onClick={() => onEventClick('linkedin')}>
        Linkedin
      </Link>
      <Link target="_blank" rel="noreferrer" href={socialLinks.TWITTER} onClick={() => onEventClick('twitter')}>
        Twitter
      </Link>
      <Link target="_blank" rel="noreferrer" href={socialLinks.EMAIL} onClick={() => onEventClick('email')}>
        Email
      </Link>
    </VStack>
  )
}
