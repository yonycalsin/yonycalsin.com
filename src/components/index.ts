export * from './achievements'
export * from './anchor'
export * from './badge'
export * from './blockquote'
export * from './breadcrumb'
export * from './button'
export * from './divider'
export * from './external-anchor'
export * from './float-socials'
export * from './heading'
export * from './icon'
export * from './illustrations'
export * from './list'
export * from './loaders'
export * from './mdx-components'
export * from './meta'
export * from './night-mode-button'
export * from './page-transition'
export * from './recommendations'
export * from './section'
export * from './spinner'
export * from './typography'
export * from './work-item'
