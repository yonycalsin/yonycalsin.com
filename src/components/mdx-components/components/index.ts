export * from './anchor'
export * from './linked-heading'
export * from './table'
