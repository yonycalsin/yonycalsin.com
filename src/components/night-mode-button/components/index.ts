export * from './moon-icon'
export * from './sun-icon'
