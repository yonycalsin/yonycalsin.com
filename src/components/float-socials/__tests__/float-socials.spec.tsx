import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { TestProvider } from 'tests'
import { FloatSocials } from 'components'

function setup() {
  return render(
    <TestProvider>
      <FloatSocials />
    </TestProvider>,
  )
}

describe('FloatSocials', () => {
  it('renders the float socials', () => {
    const view = setup()

    expect(view.container).toMatchSnapshot()
  })

  it('clicks on link', async () => {
    const view = setup()

    const github = screen.getByText(/github/i)

    await userEvent.click(github)

    const linkedin = screen.getByText(/linkedin/i)

    await userEvent.click(linkedin)

    const twitter = screen.getByText(/twitter/i)

    await userEvent.click(twitter)

    const email = screen.getByText(/email/i)

    await userEvent.click(email)

    expect(view.baseElement).toMatchSnapshot()
  })
})
