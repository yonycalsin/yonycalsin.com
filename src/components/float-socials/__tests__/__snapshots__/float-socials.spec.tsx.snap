// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatSocials clicks on link 1`] = `
<body>
  <div>
    <div
      class="space-y-5 fixed bottom-8 left-8 hidden lg:flex flex-col"
    >
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="https://github.com/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Github
      </a>
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="https://www.linkedin.com/in/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Linkedin
      </a>
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="https://twitter.com/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Twitter
      </a>
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="mailto:helloyonycalsin@gmail.com"
        rel="noreferrer"
        target="_blank"
      >
        Email
      </a>
    </div>
  </div>
</body>
`;

exports[`FloatSocials renders the float socials 1`] = `
<div>
  <div
    class="space-y-5 fixed bottom-8 left-8 hidden lg:flex flex-col"
  >
    <a
      class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
      href="https://github.com/yonycalsin"
      rel="noreferrer"
      target="_blank"
    >
      Github
    </a>
    <a
      class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
      href="https://www.linkedin.com/in/yonycalsin"
      rel="noreferrer"
      target="_blank"
    >
      Linkedin
    </a>
    <a
      class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
      href="https://twitter.com/yonycalsin"
      rel="noreferrer"
      target="_blank"
    >
      Twitter
    </a>
    <a
      class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
      href="mailto:helloyonycalsin@gmail.com"
      rel="noreferrer"
      target="_blank"
    >
      Email
    </a>
  </div>
</div>
`;
